### PR TITLE
Improve merge

### PR DESCRIPTION
**Pull request by nyamsprod**
- updated travis yaml
- update composer hook
- removed deprecated method from PublicSuffixListManager
- improve code coverage for PublicSuffixList

### CODE REVIEW

File: src/PublicSuffixListManager.php [hunk 1]
```diff
@@ -35,6 +35,15 @@ class PublicSuffixListManager
     const PRIVATE_PSL_PHP_FILE = 'private-public-suffix-list.php';
 
     /**
+     * @var PublicSuffixList Public Suffix List
```
Comment (jeremykendall) [inline src/PublicSuffixListManager.php]: Comment doesn't match variable.

File: src/PublicSuffixListManager.php [hunk 2]
```diff
@@ -35,6 +35,15 @@ class PublicSuffixListManager
...
+    protected static $domainList = [
+        self::ALL_DOMAINS => self::PDP_PSL_PHP_FILE,
+        self::ICANN_DOMAINS => self::ICANN_PSL_PHP_FILE,
+        self::PRIVATE_DOMAINS => self::PRIVATE_PSL_PHP_FILE,
+    ];
+
+    /**
      * @var string Public Suffix List URL
      */
     protected $publicSuffixListUrl = 'https://publicsuffix.org/list/effective_tld_names.dat';
```
Comment (jeremykendall) [inline src/PublicSuffixListManager.php]: We should update this to `https://publicsuffix.org/list/public_suffix_list.dat`

File: src/PublicSuffixListManager.php [hunk 3]
```diff
@@ -87,7 +127,7 @@ public function refreshPublicSuffixList()
      *
      * @return int Number of bytes that were written to the file
      */
-    public function fetchListFromSource(): int
+    protected function fetchListFromSource(): int
```
Comment (jeremykendall) [inline src/PublicSuffixListManager.php]: Nice catch. Any chance we can now change all the `protected` to `private`?

File: src/PublicSuffixListManager.php (hunk 1 above)
Comment (nyamsprod) [reply src/PublicSuffixListManager.php]: changed ... but after the merge I think the naming may still change. I used a temporary wording.

File: src/PublicSuffixListManager.php (hunk 2 above)
Comment (nyamsprod) [reply src/PublicSuffixListManager.php]: updated

**Comment by nyamsprod** [comment]
I had forgotten a typehint. The PR is complete as far as I'm concerned. 
What's missing now is the PSR-16 implementing CacheInterface to bug fix the current issue around caching.

But that's for another improvement 👍 

you could incorporate/adapt this implementation https://github.com/kodus/file-cache instead of rewriting one from scratch.

**Comment by nyamsprod** [comment]
I did no make the PublicSuffixListManager class final I'll let you decide for yourself 👍

File: src/PublicSuffixListManager.php (hunk 3 above)
Comment (nyamsprod) [reply src/PublicSuffixListManager.php]: done

**Pull request merged**